daemon/container_operations.go

```diff
@@ -678,6 +791,13 @@ func disconnectFromNetwork(container *container.Container, n libnetwork.Network,
 	}
 
 	delete(container.NetworkSettings.Networks, n.Name())
+
+	if daemon.clusterProvider != nil && n.Info().Dynamic() && !container.Managed {
```

Comment:
Do we need a nil check on `Info()`? Seem to recall a panic on a call like this recently.